packages/openid4vc/src/openid4vc-holder/OpenId4VciHolderService.ts

```diff
@@ -425,10 +492,24 @@ export class OpenId4VciHolderService {
         .withToken(tokenResponse.access_token)
 
       const credentialRequestClient = credentialRequestBuilder.build()
+
+      let createDpopOpts: CreateDPoPClientOpts | undefined
+      if (tokenResponse.dpop) {
+        const jwk = tokenResponse.dpop.dpopJwk
+        const alg = jwk.supportedSignatureAlgorithms[0]
+
+        createDpopOpts = {
+          jwtIssuer: { alg: alg as unknown as SigningAlgo, jwk: jwk.toJson() },
+          jwtPayloadProps: { accessToken: tokenResponse.access_token, nonce: tokenResponse.dpop?.dpopNonce },
+          createJwtCallback: getCreateJwtCallback(agentContext),
+        }
+      }
```

Comment:
Why not reuse `this.getCreateDPoPOptions` here? Also this could give an error when multiple alg values are supported for a jwk type, but the server only supports a specific one. So reusing the dpop alg values supported logic from existing method makes sense?

Also, accessing index [0] is unsafe again

Reply:
Also should we check if dpop is required? Can we see that using the access token? Or how do we know it is required?

Reply:
if the tokenresponse contains dpop it is required

Reply:
But token response is passed by the user

Reply:
Hmm, IMO, the user should not modify the object unless he knows what he is doing.
Also in theory it is valid to remove it since an authorization server can protect multiple credential issuers,
for some of them use of dpop may be optional even if the authorization server requires dpop.